ChangeLog

```diff
@@ -1,3 +1,158 @@
+   (LP: #1994980)
+ - tools: read-version drop extra call to git describe --long
+ - doc: improve cc_write_files doc (#1916)
+ - read-version: When insufficient tags, use cloudinit.version.get_version
+ - mounts: document weird prefix in schema (#1913)
+ - add utility function test cases (#1910) [sxt1001]
+ - test: mock file deletion in dhcp tests (#1911)
+ - Ensure network ready before cloud-init service runs on RHEL (#1893)
+   (LP: #1998655)
+ - docs: add copy button to code blocks (#1890) [s-makin]
+ - netplan: define features.NETPLAN_CONFIG_ROOT_READ_ONLY flag
+ - azure: fix support for systems without az command installed (#1908)
+ - Networking Clarification (#1892)
+ - Fix the distro.osfamily output problem in the openEuler system. (#1895)
+   [sxt1001] (LP: #1999042)
+ - pycloudlib: bump commit dropping azure api smoke test
+ - * net: netplan config root read-only as wifi config can contain creds
+ - autoinstall: clarify docs for users
+ - sources/azure: encode health report as utf-8 (#1897) [Chris Patterson]
+ - Add back gateway4/6 deprecation to docs (#1898)
+ - networkd: Add support for multiple [Route] sections (#1868)
+   [Nigel Kukard]
+ - doc: add qemu tutorial (#1863)
+ - lint: fix tip-flake8 and tip-mypy (#1896)
+ - Add support for setting uid when creating users on FreeBSD (#1888)
+   [einsibjarni]
+ - Fix exception in BSD networking code-path (#1894) [Mina Galić]
+ - Append derivatives to is_rhel list in cloud.cfg.tmpl (#1887) [Louis Abel]
+ - FreeBSD init: use cloudinit_enable as only rcvar (#1875) [Mina Galić]
+ - feat: add support aliyun metadata security harden mode (#1865)
+   [Manasseh Zhou]
+ - docs: uprate analyze to performance page [s-makin]
+ - test: fix lxd preseed managed network config (#1881)
+ - Add support for static IPv6 addresses for FreeBSD (#1839) [einsibjarni]
+ - Release 22.4.2 (#1878)
+ - status: handle ds not defined in status.json (#1876) (LP: #1997559)
+ - Make 3.12 failures not fail the build (#1873)
+ - Docs: adding relative links [s-makin]
+ - Release 22.4.1
+ - Update read-version
+ - Fix setup.py to align with PEP 440 versioning replacing trailing
+ - travis: promote 3.11-dev to 3.11 (#1866)
+ - test_cloud_sigma: delete useless test (#1828) [sxt1001]
+ - Add "nkukard" as contributor (#1864) [Nigel Kukard]
+ - tests: ds-id mocks for vmware-rpctool as utility may not exist in env
+ - net: skip duplicate mac check for netvsc nic and its VF (#1853)
+   [Anh Vo] (LP: #1844191)
+ - doc: add how to render new module doc (#1855)
+ - doc: improve module creation explanation (#1851)
+ - Add Support for IPv6 metadata to OpenStack (#1805)
+   [Marvin Vogt] (LP: #1906849)
+ - add xiaoge1001 to .github-cla-signers (#1854) [sxt1001]
+ - network: Deprecate gateway{4,6} keys in network config v2 (#1794)
+   (LP: #1992512)
+ - VMware: Move Guest Customization transport from OVF to VMware (#1573)
+   [PengpengSun]
+ - doc: home page links added (#1852) [s-makin]
+ - ChangeLog: whitespace cleanup (#1850)
```

Comment:
We have a bit of duplication that I think we should manually trim here (as we sync'd 22.4.2/22.4.1 changelog entries in separately from our 22.4.x branch.
```diff
diff --git a/ChangeLog b/ChangeLog
index 8b6d46eec..1044b7d5c 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -127,19 +127,14 @@
  - docs: uprate analyze to performance page [s-makin]
  - test: fix lxd preseed managed network config (#1881)
  - Add support for static IPv6 addresses for FreeBSD (#1839) [einsibjarni]
- - Release 22.4.2 (#1878)
- - status: handle ds not defined in status.json (#1876) (LP: #1997559)
  - Make 3.12 failures not fail the build (#1873)
  - Docs: adding relative links [s-makin]
- - Release 22.4.1
  - Update read-version
  - Fix setup.py to align with PEP 440 versioning replacing trailing
  - travis: promote 3.11-dev to 3.11 (#1866)
  - test_cloud_sigma: delete useless test (#1828) [sxt1001]
  - Add "nkukard" as contributor (#1864) [Nigel Kukard]
  - tests: ds-id mocks for vmware-rpctool as utility may not exist in env
- - net: skip duplicate mac check for netvsc nic and its VF (#1853)
-   [Anh Vo] (LP: #1844191)
  - doc: add how to render new module doc (#1855)
  - doc: improve module creation explanation (#1851)
  - Add Support for IPv6 metadata to OpenStack (#1805)
@@ -150,8 +145,6 @@
  - VMware: Move Guest Customization transport from OVF to VMware (#1573)
    [PengpengSun]
  - doc: home page links added (#1852) [s-makin]
- - ChangeLog: whitespace cleanup (#1850)
- - changelog: capture 22.3.1-4 releases
 
 22.4.2
  - status: handle ds not defined in status.json (#1876) (LP: #1997559)
```